client/src/templates/Guide/components/example/Component/index.js

```diff
@@ -0,0 +1,82 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+import PropTypes from 'prop-types';
+import { graphql } from 'gatsby';
+
+import ArticleLayout from '../../ArticleLayout';
+
+const propTypes = {
+  data: PropTypes.object
+};
+
+class InnerComponent extends React.Component {
+  state = {
+    init: false,
+    span: null,
+    div: null,
+    counter: 0
+  };
+
+  initComponent() {
```

Comment:
Would this be better handled in a `constructor`?

Reply:
It cannot be done in the constructor because of SSR.

Reply:
Understood